shared/src/vm/wasm/run.rs

```diff
@@ -243,7 +251,22 @@ where
         keys_changed,
         verifiers,
         gas_meter,
-    )
+    ) {
+        Ok(accept) => {
+            if !accept && sentinel.is_invalid_signature() {
```

Comment:
it should never be true that `accept && sentinel.is_invalid_signature()`, right? maybe we should explicitly check

Reply:
Yes correct, if the vp does verify the signature and the check on the signature failed than the vp will return `false`

Reply:
With the way we're using `verify_signatures` that would set the sentinel in our VPs the result must be `false` if the sig is invalid (there's no alternative authorization method), so I agree that we should explicitly check this